### PR TITLE
Adding some formatting and review to mesh chapter

### PR DESCRIPTION
**Pull request by DelazJ**
### Description

Goal: This is a forward port of #3562 
Add havatv suggestions
and reorganise table of contents

Ticket: https://docs.qgis.org/testing/en/docs/user_manual/working_with_mesh/mesh_properties.html

- [ ] Backport to LTR documentation is required

### Minimal requirements for merging *(for maintainers)*

- [ ] The description of this PR is coherent with the manual and does not provide wrong information.
- [ ] This PR passes the checks.

### CODE REVIEW

File: source/docs/user_manual/working_with_mesh/mesh_properties.rst [hunk 1]
```diff
@@ -11,15 +11,17 @@
...
    :maxdepth: 2
 
-Mesh properties
-===============
+What's a mesh?
+==============
 
-A mesh is an unstructured grid usually with temporal and other component . 
-The spatial component contains a collection of vertices, edges and faces in 2D or 3D space:
+A mesh is an unstructured grid usually with temporal and other components.
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: No idea. This PR simply ports changes pushed in #3562. Besides I fail to find any reference doc to mesh format...

[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: A reference for the "mesh" term would be nice, yes.  What the code seems to cover is array / vector oriented formats often used for climate data (time series, several parameters / datasets).

[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: @havatv I'm not sure I fully understand the reply.
Any suggestion to describe this or objection to merge it?

[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: No objections. We can fix this later.

[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/working_with_mesh/mesh_properties.rst]: OK.

**Pull request merged**